app/charts/bar/bars-stacked-state-props.ts

```diff
@@ -122,26 +122,26 @@ export const useBarsStackedStateData = (
 ): BarsStackedStateData => {
   const { chartConfig, observations } = chartProps;
   const { fields } = chartConfig;
-  const { x } = fields;
+  const { y } = fields;
   const {
     sortData,
-    xDimension,
+    yDimension,
     getX,
-    getXAsDate,
+    getYAsDate,
     getY,
     getSegment,
     getSegmentAbbreviationOrLabel,
     getTimeRangeDate,
   } = variables;
   const plottableData = usePlottableData(observations, {
-    getY,
+    getX,
   });
   const { sortedPlottableData, plottableDataWide } = useMemo(() => {
-    const plottableDataByX = group(plottableData, getX);
+    const plottableDataByY = group(plottableData, getY);
     const plottableDataWide = getWideData({
-      dataGroupedByX: plottableDataByX,
-      xKey: x.componentId,
-      getY,
+      dataGroupedByX: plottableDataByY,
+      xKey: y.componentId,
+      getY: getX,
       getSegment,
     });
```

Comment:
Nit: Not sure if it's worth it, but technically we could try to make the `getWideData` accept two different types to make it a bit more obvious here:

```typescript
(props:  {
  dataGroupedByX: any;
  xKey: any;
  getY: any;
  getSegment: any
} | {
  dataGroupedByY: any;
  yKey: any;
  getX: any;
  getSegment: any
})
```

but it would probably make the inner implementation more complex as we'd need to check if one or the other is present 🤔

Reply:
Also wondering if `getWideDataInverted` shouldn't be used here, as it's used down the road in stacked bars state? And maybe it could be named `getHorizontalWideData` (with `getVerticalWideData` for the current `getWideData`)?

I think it would be a bit more clear, as now we have a "baseline" function and an inverted one, but we don't know what's exactly inverted from the function names 🤔